meinberlin/apps/contrib/assets/dropdown.js

```diff
@@ -1,41 +1,67 @@
-const Dropdown = {
-  toggleDropdown (event) {
-    const dropdown = event.currentTarget.parentNode
-    const isOpen = dropdown.getAttribute('aria-expanded') === 'true'
+class Dropdown {
+  constructor (container) {
+    this.container = container
+    this.trigger = container.querySelector('[data-dropdown-trigger]')
+    this.dropdown = container.querySelector('[data-dropdown-menu]')
+    this.closeTimeout = null
 
-    dropdown.setAttribute('aria-expanded', !isOpen)
+    this.init()
+  }
 
-    const menu = dropdown.querySelector('.dropdown__menu')
-    menu.classList.toggle('dropdown__menu--show', !isOpen)
+  init () {
+    this.trigger.addEventListener('click', (event) => this.toggleDropdown(event))
+    this.dropdown.addEventListener('keyup', (event) => this.handleKeyup(event))
+    this.dropdown.addEventListener('focusout', (event) => this.handleFocusout(event))
+  }
 
-    if (!isOpen) {
-      const menu = dropdown.querySelector('.dropdown__menu')
-      menu.firstElementChild.focus()
+  toggleDropdown (event) {
+    event.preventDefault()
+    if (this.container.classList.contains('is-open')) {
+      this.closeDropdown()
+    } else {
+      this.openDropdown()
     }
-  },
+  }
+
+  openDropdown () {
+    this.dropdown.style.display = 'block'
```

Comment:
I'm fine with this but just wondering if having it in the css is a bit of a clearer division between functionality and styling as we have the `is-open` class to apply it ?